src/main/java/sparrow/data/task/TaskList.java

```diff
@@ -0,0 +1,91 @@
+package sparrow.data.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Encapsulates task list with operations to be carried out on list.
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Constructs an empty task list.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a task list with the input list.
+     * @param tasks input task list
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
+    public void addTask(Task toAdd) {
+        tasks.add(toAdd);
+    }
+    /**
+     * Returns a task list matching the inputted date filter.
+     * @param dateFilter User's command and date to filter by.
+     * @return Filtered list based on input.
+     * @throws DateTimeParseException if date passed cannot be converted to a LocalDate object.
+     */
+    public ArrayList<Task> filterList(LocalDate dateFilter) throws DateTimeParseException {
+        ArrayList<Task> filteredList = new ArrayList<>();
+
+        for (Task task : tasks) {
+            LocalDate taskDate;
+
+            // Get date from Deadline/Event
+            if (task instanceof Deadline) {
+                taskDate = ((Deadline) task).getDueDate();
+            } else if (task instanceof Event) {
+                taskDate = ((Event) task).getDate();
+            } else {
+                continue;
+            }
+
+            // Check if task's date is on date specified
+            if (taskDate.isEqual(dateFilter)) {
+                filteredList.add(task);
+            }
+        }
+        return filteredList;
+    }
+
+    /**
+     * Marks a task in the task list as completed.
+     * @param taskNum Task number corresponding to completed task.
```

Comment:
I think you can add a line in between the parameters and the description of the method